core/embed/sec/secret/inc/sec/secret_keys.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * This file is part of the Trezor project, https://trezor.io/
+ *
+ * Copyright (c) SatoshiLabs
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <trezor_types.h>
+
+#ifdef SECURE_MODE
+
+#ifdef USE_OPTIGA
+secbool secret_key_optiga_get(uint8_t* dest, size_t len);
+#endif
+
+#ifdef USE_TROPIC
+secbool secret_key_tropic_tropic_pubkey_get(uint8_t* dest, size_t len);
+
+secbool secret_key_tropic_trezor_privkey_get(uint8_t* dest, size_t len);
```

Comment:
We discussed it with @andrewkozlik and we came up with this API:

```
#define OPTIGA_PAIRING_SECRET_SIZE 32
secbool secret_key_optiga_pairing(uint8_t dest[OPTIGA_PAIRING_SECRET_SIZE]);

secbool secret_key_tropic_pairing_public(curve25519_key dest);
secbool secret_key_tropic_pairing(curve25519_key dest);
```

The change we consider most important is the removal of the `len` parameter.

What do you think about it?

Reply:
see https://github.com/trezor/trezor-firmware/commit/312cf75311445b572016ca85d197030b0a0d3025